src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -298,6 +298,74 @@ public override bool HasDefaultValue
             return defaultValue;
         }
 
+        private bool ParameterTypeIsDateTime { get => ParameterType == typeof(DateTime) || ParameterType == typeof(DateTime?); }
+
+        private CustomAttributeData? GetPrioritizedCustomAttributeDataForRawDefaultValueResolution(IEnumerable<CustomAttributeData> customAttributes)
+        {
+            CustomAttributeData? previousAttributeData = null;
+            foreach (CustomAttributeData currentAttributeData in customAttributes)
+            {
+                Type currentAttributeType = currentAttributeData.AttributeType;
+                if (ParameterTypeIsDateTime && currentAttributeType == typeof(DateTimeConstantAttribute))
+                {
+                    previousAttributeData = currentAttributeData;
+                    break;
+                }
+                else if (previousAttributeData == null && (currentAttributeType.IsSubclassOf(typeof(CustomConstantAttribute)) || currentAttributeType == typeof(DecimalConstantAttribute)))
```

Comment:
I would expect both `DateTimeConstantAttribute` and `DecimalConstantAttribute` ordering to be special-cased same way. Why we are special-casing them differently?

Reply:
I tried to match what coreCLR was doing before i.e. prioritising `DateTimeConstantAttribute` over `CustomConstantAttribute` and `DecimalConstantAttribute`. Regarding special-casing the `DecimalConstantAttribute`, I am fine with having the same rules as for DateTime parameter types, but at the same time, I was concerned of changing the existing implementation too much.

Reply:
Should I adapt the rules to special-case `DecimalConstantAttribute` in the same fashion?

Reply:
I am not sure what you mean exactly. I would try to make this as simple and as consistent as possible.

Reply:
One possible variant:
```
foreach (CustomAttributeData attributeData in customAttributes)
{
     Type attributeType = attributeData.AttributeType;
     if (attributeType == typeof(DecimalConstantAttribute))
     {
         return GetRawDecimalConstant(attr);
     }
     if (attributeType.IsSubclassOf(typeof(CustomConstantAttribute))
     {
         if (attributeType == typeof(DateTimeConstantAttribute)
         {
             return GetRawDateTimeConstant(attributeData);
         }
         return GetRawConstant(attributeData);
     }
}
return DBNull.Value;
```

Reply:
My latest changes use this approach i.e. "use the first attribute attached to the parameter which provides a constant value" for both raw and non-raw default value resolution for coreCLR.